File: cases/taint/csharp/local-chain-positive/LocalChainPositive.cs
namespace DataFlowBench;

static class LocalChainPositive
{
    static int dfb_source() // DFB-SOURCE: local-chain-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: local-chain-sink

    static void Run()
    {
        int first = dfb_source();
        int second = first; // DFB-WITNESS: local-chain-second
        int third = second; // DFB-WITNESS: local-chain-third
        dfb_sink(third);
    }
}

File: cases/taint/rust/local-chain-positive/local_chain_positive.rs
fn dfb_source() -> i32 { // DFB-SOURCE: local-chain-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: local-chain-sink

fn run() {
    let first = dfb_source();
    let second = first; // DFB-WITNESS: local-chain-second
    let third = second; // DFB-WITNESS: local-chain-third
    dfb_sink(third);
}
